src/dev/build/tasks/nodejs_modules/code_parser/code_parser.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { canRequire } from './can_require';
+import { read } from '../../../lib';
+import { extname } from 'path';
+import * as parser from '@babel/parser';
+import traverse from '@babel/traverse';
+
+async function parseSingleFile(filePath, visitorsGenerator) {
+  const results = [];
+
+  // Don't parse any other files than .js ones
+  if (extname(filePath) !== '.js') {
+    return results;
+  }
+
+  // Read the file
+  const content = await read(filePath, { encoding: 'utf8' });
+
+  // Parse and get the code AST
+  // All the babel parser plugins
+  // were enabled
+  const ast = parser.parse(content, {
+    sourceType: 'unambiguous',
+    plugins: [
+      'asyncGenerators',
+      'classProperties',
+      'dynamicImport',
+      'exportDefaultFrom',
+      'exportNamespaceFrom',
+      'objectRestSpread',
+      'throwExpressions'
```

Comment:
Should we get these from wherever we have the babel preset defined so we don't have to change in two places?

Reply:
I think we will need to pass that Josh. Those are plugin names in order to configure the babel parser options (https://babeljs.io/docs/en/babel-parser#plugins). For the common babel configs we are relying in our internal `kbn-babel-preset` to retrieve the presets, but presets are different from those plugins here. Do you think it does make sense to add a new file called for example `common_babel_parser_options.js` to the `kbn-babel-preset` and just load the parser configuration from here?

Reply:
That seems like a good idea. I think it will be pretty hard to find this whenever we need change these and could lead to some confusion/bugs.

Reply:
done